.github/workflows/demo_check.yml

```diff
@@ -0,0 +1,186 @@
+            -S -s -nographic                                                                                  \
+            &
+
+          gdb-multiarch --batch                                                                               \
+            -ex "set output-radix 16"                                                                         \
+            -ex "target remote :1234"                                                                         \
+            -ex "watch global"                                                                                \
+            -ex "continue"                                                                                    \
+            -ex "monitor quit"                                                                                \
+            ${{ github.workspace }}/${{ steps.trim.outputs.trimmed }}/Debug/single_level.elf                  \
+            > gdb.log
+
+      - name: Check Single Level Result
+        run: |
+          cat gdb.log
+        
+          grep -q "New value = 0x5" gdb.log
+
+  check_multiple_inheritance:
+    runs-on: ubuntu-24.04
+    needs: build_debug
+    steps:
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install qemu-system-arm gdb-multiarch
+          
+      - name: Trim profile version
+        id: trim
+        run: |
+          profile="${{ inputs.platform_profile }}"
+          echo "trimmed=${profile##*/}" >> $GITHUB_OUTPUT
+      
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-${{ steps.trim.outputs.trimmed }}
+
+      - name: Run Multiple Inheritance Demo
+        run: |
+          nohup qemu-system-arm                                                                                 \
+            -M olimex-stm32-h405                                                                                \
+            -kernel ${{ github.workspace }}/${{ steps.trim.outputs.trimmed }}/Debug/multiple_inheritance.elf    \
+            -S -s -nographic                                                                                    \
+            &
+
+          gdb-multiarch --batch                                                                                 \
+            -ex "set output-radix 16"                                                                           \
+            -ex "target remote :1234"                                                                           \
+            -ex "watch global"                                                                                  \
+            -ex "continue"                                                                                      \
+            -ex "monitor quit"                                                                                  \
```

Comment:
What happens if we never update global? Is there a way to set a timeout with continue? Otherwise our CI may get stuck and we end up wasting a lot of server time. Same goes for  the other test.

Reply:
We could add the `timeout` command before the `gdb-multiarch` and that should do that for us. I'm thinking like 5 seconds is enough.